### PR TITLE
MINIFICPP-1910 - Rename main directory to avoid name clashes with main branch

### PR DESCRIPTION
**Pull request by adam-markovics**
https://issues.apache.org/jira/browse/MINIFICPP-1910

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by lordgamez** [comment]
It probably also have to be changed in the `msi/WixWin.wsi` and `msi/WixWinMergeModules.wsi` files as well where the installed binary's source is the `main` directory set to `main\minifi.exe`

**Comment by adam-markovics** [comment]
> It probably also have to be changed in the `msi/WixWin.wsi` and `msi/WixWinMergeModules.wsi` files as well where the installed binary's source is the `main` directory set to `main\minifi.exe`. The same directory is also mentioned in the Windows.md documentation

Thanks for the Windows-specific update. If someone has found necessary changes for other OSes and distributions, please share.

**Comment by szaszm** [comment]
You can use `--` to disabmiguate.
`git checkout main --` means switch to the main branch.
`git checkout -- main` means restore the main directory.

**Comment by adam-markovics** [comment]
> You can use `--` to disabmiguate. `git checkout main --` means switch to the main branch. `git checkout -- main` means restore the main directory.

I have been doing this all the time, it would just be more comfortable if I didn't have to.

**Review by fgerlits — APPROVED** [review]

**Review by lordgamez — APPROVED** [review]

File: win_build_vs.bat [hunk 1]
```diff
@@ -108,9 +108,9 @@ mkdir %builddir%
 pushd %builddir%\
 
 if [%generator%] EQU ["Ninja"] (
-    set "buildcmd=ninja && copy bin\minifi.exe main\"
+    set "buildcmd=ninja && copy bin\minifi.exe minifi_main\"
 ) else (
-    set "buildcmd=msbuild /m nifi-minifi-cpp.sln /property:Configuration=%cmake_build_type% /property:Platform=%build_platform% && copy bin\%cmake_build_type%\minifi.exe main\"
+    set "buildcmd=msbuild /m nifi-minifi-cpp.sln /property:Configuration=%cmake_build_type% /property:Platform=%build_platform% && copy bin\%cmake_build_type%\minifi.exe minifi_main\"
```
Comment (adamdebreceni) [inline win_build_vs.bat:113]: what is the purpose of the copy command here? is it required to run the tests?

Comment (fgerlits) [reply win_build_vs.bat:113]: It is required by `cpack`, when creating the `msi` installer.

Comment (adamdebreceni) [reply win_build_vs.bat:113]: I see, thanks

**Review by adamdebreceni — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**